### PR TITLE
fix: return data

### PR DESCRIPTION
**Pull request by terrarium789**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -50,22 +57,49 @@ export function useFlexSearch <T extends {id: string| number}> (
...
-        return store.value.find(storeItem => storeItem.id === rawResult)
       })
     }
   })
 }
+function isSimpleDocumentSearchResultSetUnit (
+  object: FlexSearch.Id | FlexSearch.SimpleDocumentSearchResultSetUnit
+): object is FlexSearch.SimpleDocumentSearchResultSetUnit {
+  return (
+    (object as FlexSearch.SimpleDocumentSearchResultSetUnit).result !==
```
Comment (50rayn) [inline src/index.ts]: @LwveMike Is this ok in you opinion?

File: src/index.ts [hunk 2]
```diff
@@ -23,15 +23,22 @@ export function useFlexSearch <T extends {id: string| number}> (
   watch(
```
Comment (50rayn) [inline src/index.ts]: @terrarium789 Changes in this _watch_ are not reasonable. Please revert changes from this function

File: src/index.spec.ts [hunk 3]
```diff
@@ -28,7 +29,7 @@ function dummyDataUser () {
   const providedIndex = ref(dummyDocumentIndex(['firstName', 'lastName', 'email']))
 
   for (let i = 0; i < faker.datatype.number(); i++) {
-    const user = { firstName: faker.name.firstName(), lastName: faker.name.lastName(), id: i }
+    const user = { firstName: faker.phone.number(), lastName: faker.phone.imei(), id: i }
```
Comment (50rayn) [inline src/index.spec.ts]: Using `faker.phone` for _firstName_ and _lastName_ is not ok.
Changes the object structure to `{ fullName: faker.name.fullName, email: faker.internet.email(), phone: faker.phone.number(), id: i }`

**Review by 50rayn — CHANGES_REQUESTED** [review]
* Some unreasonable changes in the watcher, that should be reverted 
* Some unlogical property nameing in test

File: src/index.ts (hunk 1 above)
Comment (LwveMike) [reply src/index.ts]: Using 'as' in a type predicate defeats its purpose.

File: src/index.spec.ts [hunk 4]
```diff
@@ -28,7 +36,11 @@ function dummyDataUser () {
   const providedIndex = ref(dummyDocumentIndex(['firstName', 'lastName', 'email']))
 
   for (let i = 0; i < faker.datatype.number(); i++) {
-    const user = { firstName: faker.name.firstName(), lastName: faker.name.lastName(), id: i }
+    const user = {
+      firstName: faker.name.firstName() + faker.random.word(),
+      lastName: faker.name.lastName() + faker.random.word(),
+      id: i
```
Comment (romanbitca) [inline src/index.spec.ts]: 'id' should be before 'lastName'

File: src/index.ts (hunk 2 above)
Comment (romanbitca) [reply src/index.ts]: agree, better with "{" and "}".

**Review by 50rayn — APPROVED** [review]

**Comment by LwveMike** [comment]
Approved

**Review by LwveMike — APPROVED** [review]

**Review by romanbitca — APPROVED** [review]

**Pull request merged**